### PR TITLE
[MM-44004] Optionally turn off UTF-8 encoding for winreg if the system doesn't support it

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
It's possible that some system administrators may have `cmd.exe` disabled in a more controlled environment, but still might want to configure their Mattermost servers using GPO.

Our current library requires the use of `cmd.exe` to get the registry entries since some users use special characters which require UTF-8 encoding to add correctly to the app. This conflicts with what some others might want.

This PR adds some error-checking logic that tries to initialize the registry object with UTF-8 encoding disabled if it fails the first time.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44004

```release-note
Disable UTF-8 encoding for systems that don't support it correctly
```

### CODE REVIEW

**Comment by coltoneshaw** [comment]
@hmhealey - I know Devin is out right now, but do you happen to know if there is a way to get a build of the `msi` installer for this?

**Comment by coltoneshaw** [comment]
Actually, I can probably add the `msi` flag to the electron builder config, but i don't see it already so i'm wondering if we build it through another process that i'm missing.

**Comment by hmhealey** [comment]
No, sorry. I don't have any experience with building the desktop app, particularly when it comes to building for Windows

**Comment by devinbinnie** [comment]
@coltoneshaw Here you go!

https://output.circle-artifacts.com/output/job/5e9ea70a-a737-4269-83b3-937bedae1773/artifacts/0/packages/win-release/5.1.0-develop.1/mattermost-desktop-5.1.0-develop.1-x64.msi

**Comment by coltoneshaw** [comment]
@devinbinnie - The customer was able to confirm here that this specific issue still exists with the latest version. It's attempting to call `cmd.exe` and getting blocked.

**Comment by devinbinnie** [comment]
@coltoneshaw Can you have them send over the Desktop App logs? Also, make sure they're using the correct version, since the MSI installer installs to a different location than the EXE

**Comment by coltoneshaw** [comment]
@devinbinnie - Of course, i'll see if I can get the logs and share them privately. The version should be right because they uninstalled the prior version and installed this one.

**Comment by devinbinnie** [comment]
@coltoneshaw Any progress on this from the customer?

**Comment by coltoneshaw** [comment]
Just a quick update, and I'm going to play around with changes.

I can repro this 100% of the time now. 
1. Open Local Group Policy Editor
2. User config > Admin Templates > system > Prevent access to the command prompt
3. Set that to enabled, and block script access too.
4. Set some group policy things, I did the default server.
5. Open Mattermost and you won't see the default server
6. Go in and enable cmd access again (step 2)
7. open mattermost and you'll see the default server.
8. Repeat as many times as you want.

**Comment by devinbinnie** [comment]
@coltoneshaw I added some changes. Turns out the `winreg-utf8` library won't work regardless of whether you turn UTF-8 on or off, so I've elected to just toggle between the two libraries and will enforce the requirement that if you need UTF-8 enabled server names then you can't disable the command prompt.

**Comment by coltoneshaw** [comment]
@devinbinnie - Thanks! Let me test it and let you know

**Comment by coltoneshaw** [comment]
@devinbinnie - On a test with `cmd.exe` fully blocked the app appears to still not read the registry. If i enable `cmd.exe` the app works fine. I'm testing with the above repro steps.

**Review by m1lt0n — APPROVED** [review]
Looks good!

File: src/common/config/RegistryConfig.ts [hunk 1]
```diff
@@ -148,4 +161,12 @@ export default class RegistryConfig extends EventEmitter {
             }
         });
     }
+
+    createRegistry(hive: string, key: string, utf8 = true) {
+        if (utf8) {
+            return new WindowsRegistryUTF8({hive, key, utf8});
+        }
+
+        return new WindowsRegistry({hive, key});
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/common/config/RegistryConfig.ts:170]: It ends up doing the exact same thing as `winreg-utf8` and still would have issues when trying to access `cmd.exe`

**Comment by coltoneshaw** [comment]
@devinbinnie - I think I just replicated the customers behavior using applocker and can confirm it's still broken

**Comment by coltoneshaw** [comment]
[desktoplogs.log](https://github.com/mattermost/desktop/files/10449924/desktoplogs.log)

@devinbinnie - Attached are the logs for using applocker.

To repro I followed the steps here - https://winbuzzer.com/2021/09/21/how-to-configure-applocker-in-windows-10-to-block-a-script-from-running-xcxwbt/
Screenshot 2023-01-18 at 2 25 39 PM

Screenshot 2023-01-18 at 2 25 47 PM
Screenshot 2023-01-18 at 2 25 59 PM
Screenshot 2023-01-18 at 2 26 12 PM

**Comment by coltoneshaw** [comment]
One note, when you enable those rules it'll block a lot of things so be careful. Tried to tone them down a bit but I'm still learning applocker. If you want to test with them off you have to delete the rules and restart to get Mattermost to read the registry.

I've enabled them, Mattermost doesn't see my registry config. Disabled (restarted and everything) and Mattermost works fine.

**Pull request merged**